lib/handlers/post.js

```diff
@@ -80,20 +81,19 @@ function handler (req, res, next) {
 
   function one () {
     debug('Receving one file')
-    var linkHeader = header.parseMetadataFromHeader(req.get('Link'))
-    var slug = req.get('Slug')
-    ldp.post(
-      req.hostname,
-      containerPath,
-      slug,
-      req,
-      linkHeader.isBasicContainer,
+    const { slug, link, 'content-type': contentType } = req.headers
+    const links = header.parseMetadataFromHeader(link)
+    const mimeType = contentType ? contentType.replace(/\s*;.*/, '') : ''
+    const extension = mimeType in extensions ? `.${extensions[mimeType][0]}` : ''
```

Comment:
Ditto here.

Reply:
Actually this line is really straightforward. Extension is a hash with non-empty arrays of extensions; so either it's in there and I return the first, or it isn't and I return the empty string. Testing this line in isolation would be just testing the `extensions` data structure, so that doesn't seem really meaningful (unless you meant something else than the line).

Reply:
Then again, I suppose I could put both lines in a `util` library. Is this what you meant?

Reply:
Hope that my response above clarifies my thought.